rclpy/src/rclpy_common/include/rclpy_common/handle.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY_COMMON__HANDLE_H_
+#define RCLPY_COMMON__HANDLE_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <Python.h>
+#include <stddef.h>
+
+#include "rcutils/types/rcutils_ret.h"
+
+/// Wrapper that manages the lifetime of another object,
+/// allowing to establish dependencies between them.
+typedef struct rclpy_handle_t rclpy_handle_t;
+
+/// Signature of rclpy_handle_t destructors.
+typedef void (* rclpy_handle_destructor_t)(void *);
+
+/// Create a PyCapsule wrapping a rclpy_handle_t object.
+/**
+ * \sa _rclpy_create_handle
+ * \param name Name of the PyCapsule.
+ * \returns PyCapsule wrapping the rclpy_handle_t.
+ */
+PyObject *
+rclpy_create_handle_capsule(void * ptr, const char * name, rclpy_handle_destructor_t destructor);
+
+/// Returns the object managed by the rclpy_handle_t wrapped in a PyCapsule.
+void *
+rclpy_handle_get_pointer_from_capsule(PyObject * capsule, const char * name);
+
+/// Creates a rclpy_handle_t object.
+/**
+ * \param ptr Opaque pointer to the object being wrapped.
+ * \param destructor Function that will be called when the handle is destructed.
+ */
+rclpy_handle_t *
+_rclpy_create_handle(void * ptr, rclpy_handle_destructor_t destructor);
+
+/// Create a PyCapsule wrapping a rclpy_handle_t object.
+/**
+ * \param rclpy_handle_t Already constructed handle.
+ * \param name Name of the PyCapsule.
+ * \returns PyCapsule wrapping the rclpy_handle_t, using _rclpy_handle_dec_ref as destructor.
+ */
+PyObject *
+_rclpy_create_handle_capsule(rclpy_handle_t * ptr, const char * name);
+
+/// Returns the object managed by the rclpy_handle_t.
+/**
+ * PyExc_RuntimeError is set, if `handle` is NULL, o if the managed pointer is NULL.
+ *
+ * \param handle
+ */
+void *
+_rclpy_handle_get_pointer(rclpy_handle_t * handle);
+
+/// Adds a dependency to a handle.
+/**
+ * PyExc_RuntimeError is set, if `handle` is NULL, o if the managed pointer is NULL.
+ *
+ * \param dependency Handle object which its reference count will be incremented.
```

Comment:
@ivanpauno consider 
```suggestion
 * \param dependency Handle object whose reference count will be incremented.
```

Reply:
Solved in https://github.com/ros2/rclpy/pull/497/commits/034da37e83c05acab31b7efc4436aa55fb36e08b.